discojs/discojs-node/src/models/model_loader.ts

```diff
@@ -0,0 +1,23 @@
+import fs from 'node:fs'
+import fsPromises from 'node:fs/promises'
```

Comment:
async all the way! there is a bunch of [TOCTOU](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use) to otherwise.

```suggestion
import fs from 'node:fs/promises'
```